akka-javasdk/src/main/scala/akka/javasdk/impl/agent/AgentImpl.scala

```diff
@@ -125,10 +130,15 @@ private[impl] final class AgentImpl[A <: Agent](
             }
             val spiModelProvider = toSpiModelProvider(req.modelProvider)
             val metadata = MetadataImpl.toSpi(req.replyMetadata)
+
+            // FIXME refactor to persist both user and ai message on transform method
```

Comment:
yes, that is not only more efficient, but also important for the error case, when we don't want to save the user message